src/main/java/com/karankumar/bookproject/backend/service/BookService.java

```diff
@@ -57,13 +66,24 @@ public void save(Book book) {
     private boolean bookHasAuthorAndPredefinedShelf(Book book) {
         return book != null && book.getAuthor() != null && book.getPredefinedShelf() != null;
     }
+    
+    private boolean bookHasPublisher(Book book) {
+    	return book.getPublisher() != null;
+    }
 
     private void addBookToAuthor(Book book) {
         Author author = book.getAuthor();
         Set<Book> authorBooks = author.getBooks();
         authorBooks.add(book);
         author.setBooks(authorBooks);
     }
+    
+    private void addBookToPublisher(Book book) {
+        Publisher publisher = book.getPublisher();
+        Set<Book> publisherBooks = publisher.getBooks();
+        publisherBooks.add(book);
+        publisher.setBooks(publisherBooks);
+    }
```

Comment:
I feel this is the job of publisher service so can we please move it there? Also can you please add a test for the above logic?